JustFakeIt/MatchingStrategy/BodyMatching/PartialJsonMatching.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace JustFakeIt
+{
+    public class PartialJsonMatching : IBodyMatchingOption
+    {
+        public bool MatchBody(string expectedBody, string actualBody)
+        {
+            return string.IsNullOrEmpty(expectedBody) || actualBody.Equals(expectedBody) || JsonKeyValuesMatch(expectedBody, actualBody);
+        }
+
+        public bool JsonKeyValuesMatch(string expectedBody, string actualBody)
+        {
+            JObject expectedJObject;
+            JObject actualJObject;
+
+            try
+            {
+                expectedJObject = JsonConvert.DeserializeObject<JObject>(expectedBody);
```

Comment:
The reason was to use a try catch to make sure the expected and actual are valid json. If they are not exception is thrown and we deal w it by leaving the recursion and flagging the match as a false.